DataFormats/Math/interface/libminifloat.h

```diff
@@ -170,6 +127,14 @@ class MiniFloatConverter {
   }
 
 private:
+  //in C++20 we can use std::bit_cast which is constexpr
```

Comment:
is it worthwhile to fold this in `#ifndef __cpp_lib_bit_cast`  and `#include <bit>` at the top if it's defined?

Reply:
I took a look at https://en.cppreference.com/w/cpp/feature_test and it seems to me like the feature test system is really meant for libraries which are meant to support many different C++ versions simultaneously. That isn't what we really do. Once we validate C++20 option for our compilers (which we haven't even started since no compiler supports all the items we might want) we will not need to compile that code with an older compiler.

So personally, I don't see a compelling reason to add 10+ lines of code (as a guess) which are only useful during the short window between the C++17 to C++20 transition. If you really want that ability, I will change the code.

Reply:
@slava77 are you 'OK' with my reasoning about not using the test feature or would you prefer them to be added to the code?

Reply:
I have a more broad question: what are we going to do with all other use cases of union?
e.g. DataFormats/Math/interface/FastMath.h, SIMDVec.h, approx_math.h

the bit decoding is also used quite extensively in DataFormats/GEMDigi, e.g. interface/GEMVFAT.h; is that part OK, or does it need to switch to memcpy as well?

Reply:
@slava77 I'm of the opinion that we should change all the uses of union for bit decoding eventually. It would be nice to have the `bit_cast` for that but I'd be OK with doing the `memcpy`. 

The only place we discovered `memcpy` to be a problem is when used with the NVidia compiler as that one still calls the `memcpy` function rather than replacing it with inlined machine code. So for GPU shared code, we have to wait for the `bit_cast` support (in gcc/clang and ncc) before we can remove the undefined behavior.

Reply:
is this PR with a local definition of `bit_cast` the only one required now until we get to C++20, or will more going to be required/made soon?
If it's the latter, it sounds like making a DataFormats/Math/interface/bit_cast.h to handle this is a more appropriate solution.

Reply:
I was thinking that just calling `memcpy` in the other cases directly was how I was going to handle it. I first tried it that way here, but there was so much repetition that I switched to defining a `bit_cast` (which drastically simplified the code, surprisingly).

If we were to have our own `bit_cast` it would more likely be put in `FWCore/Utilities` since we've often put `soon to come to std` items there in the past.

Reply:
As for the other places in `DataFormat/Math` I'm not sure the compilers are smart enough to be able to vectorize the replacements for `memcpy` they use. So I'd skip those for now.

Reply:
Do you want us to make the `bit_cast` separately for this pull request or can we wait until we see a need?

Reply:
> Do you want us to make the `bit_cast` separately for this pull request or can we wait until we see a need?

from https://github.com/cms-sw/cmssw/pull/35054#discussion_r700435824
I concluded that there is a need (there just isn't another PR yet that addresses other use cases of a union).